build/VersionHelper.cs

```diff
@@ -0,0 +1,24 @@
+using System.Reflection;
+
+public static class VersionHelper
```

Comment:
Should we use something like https://learn.microsoft.com/en-us/dotnet/api/system.version.tryparse?view=net-7.0 to help us instead of the manual string parsing that we are doing here?

Reply:
`Version.Parse` is not working correctly with format we need, eg `4.2.1-rc.1.54` or even `4.2.1-rc.1.54+commithashhere`.

Technically I could rewrite GetVersionsParts method, but I do not see big benefits to parse it to ints.